### PR TITLE
Retrieve credentials for saved conversations

### PR DESCRIPTION
**Pull request by maxdeviant**
This PR fixes an issue where credentials for saved conversations weren't being fully retrieved, which was preventing continuing a conversation with the assistant in a saved conversation.

Release Notes:

- Fixed continuing a saved conversation with the Zed assistant.

### CODE REVIEW

File: crates/assistant/src/assistant_panel.rs [hunk 1]
```diff
@@ -1507,7 +1508,9 @@ impl Conversation {
             )
             .await,
         );
-        cx.update(|cx| completion_provider.retrieve_credentials(cx))?;
+        cx.update(|cx| completion_provider.retrieve_credentials(cx))?
+            .await;
```
Comment (maxdeviant) [inline crates/assistant/src/assistant_panel.rs:1512]: We weren't `await`ing the call to retrieve the credentials 🤦🏻‍♂️

**Review by nathansobo — COMMENTED** [review]
Whoops! Thank you!

File: crates/assistant/src/assistant_panel.rs (hunk 1 above)
Comment (SomeoneToIgnore) [reply crates/assistant/src/assistant_panel.rs:1512]: Would `#[must_use]` help if placed on `BoxFuture` & friends?

Comment (maxdeviant) [reply crates/assistant/src/assistant_panel.rs:1512]: I was wondering that.

It looks like `BoxFuture` is coming from the `futures` crate, so we'd either have to wrap it or try to upstream.

Comment (SomeoneToIgnore) [reply crates/assistant/src/assistant_panel.rs:1512]: I've done a blasphemous direct update of `.cargo/registry/src` file with the `BoxFuture` and it does not seem to help, alas: with or without this `.await`, the `cargo clean & cargo check --all --all-targets` output did not change.

Comment (SomeoneToIgnore) [reply crates/assistant/src/assistant_panel.rs:1512]: And as @\osiewicz noted, `cargo +nightly check` does detect such issues: it seems there was exactly one place without `.await` in the project currently.

**Pull request merged**